### PR TITLE
rust: add support to ethtool

### PR DESCRIPTION
**Pull request by ffmancera**
Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

File: automation/run-tests.sh [hunk 1]
```diff
@@ -259,6 +259,15 @@ function run_tests {
...
             not test_explicitly_ignore_a_bridge_port' \
             ${nmstate_pytest_extra_args}"
+        exec_cmd "
+          env  \
+          PYTHONPATH=$CONTAINER_WORKSPACE/rust/src/python \
+          pytest \
+            $PYTEST_OPTIONS \
+            tests/integration/ethtool_test.py \
+            -k '\
+            not test_ethtool_invalid_feature' \
```
Comment (cathay4t) [inline automation/run-tests.sh]: Maybe we should change this test case to expect `NmstateValueError` as we should not allowing arbitrary feture string. The failure should happen at validation stage.

The `#[serde(deny_unknown_fields)]` can do the magic.

File: rust/src/lib/ifaces/ethtool.rs [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    pub tx_generic_segmentation: Option<bool>,
+}
+
+impl EthtoolFeatureConfig {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Eq, PartialEq, Clone, Default)]
```
Comment (cathay4t) [inline rust/src/lib/ifaces/ethtool.rs]: This is struct only bool and u32, you may enable `Copy` trait which could make the life easier as you don't need to worry about ownership of it.

**Review by cathay4t — APPROVED** [review]
Looks good to me!

File: automation/run-tests.sh (hunk 1 above)
Comment (ffmancera) [reply automation/run-tests.sh]: Done, thanks!

File: rust/src/lib/ifaces/ethtool.rs (hunk 2 above)
Comment (ffmancera) [reply rust/src/lib/ifaces/ethtool.rs]: Done, thank you!

**Pull request merged**